model/ProctorManagementService.php

```diff
@@ -0,0 +1,162 @@
+    }
+
+
+    /**
+     * Allow test center administrator to authorize multiple proctors to test centers
+     * @param $proctorsUri
+     * @param $testCenters
+     * @return bool
+     */
+    public function authorizeProctors($proctorsUri, $testCenters){
+        $return = true;
+        $propertiesValues = array(self::PROPERTY_AUTHORIZED_PROCTOR_URI => $testCenters);
+        foreach($proctorsUri as $proctorUri){
+            $proctor = new core_kernel_classes_Resource($proctorUri);
+            $return &= $proctor->setPropertiesValues($propertiesValues);
+        }
+
+        return (bool) $return;
+    }
+
+    /**
+     * Allow test center administrator to remove authorization to multiple proctors
+     * @param $proctorsUri
+     * @param $testCenters
+     * @return bool
+     */
+    public function unauthorizeProctors($proctorsUri, $testCenters){
+        $return = true;
+        foreach($proctorsUri as $proctorUri){
+            $proctor = new core_kernel_classes_Resource($proctorUri);
+            foreach($testCenters as $testCenter){
+                $return &= $proctor->removePropertyValue(new core_kernel_classes_Property(self::PROPERTY_AUTHORIZED_PROCTOR_URI), $testCenter);
+
+            }
+        }
+
+        return (bool) $return;
+    }
+
+    /**
+     * Allow test center administrator to assign multiple proctors to test centers
+     * @param $proctorsUri
+     * @param $testCenterAdminUri
+     * @return bool
+     */
+    public function assignProctors($proctorsUri, $testCenterAdminUri){
+        $return = true;
+        $testCenterAdmin = new core_kernel_classes_Resource($testCenterAdminUri);
+        $testCenters = $testCenterAdmin->getPropertyValues(new core_kernel_classes_Property(self::PROPERTY_ADMINISTRATOR_URI));
+        $propertiesValues = array(self::PROPERTY_ASSIGNED_PROCTOR_URI => $testCenters);
+        foreach($proctorsUri as $proctorUri){
+            $proctor = new core_kernel_classes_Resource($proctorUri);
+            $return &= $proctor->setPropertiesValues($propertiesValues);
+        }
+
+        return (bool) $return;
+    }
+
+    /**
+     * Return all proctor assigned to test centers managed by an admin
+     * @param string $testCenterAdminUri
+     * @return User[]
+     */
+    public function getAssignedProctors($testCenterAdminUri){
+        $testCenterAdmin = new core_kernel_classes_Resource($testCenterAdminUri);
+        $testCenters = $testCenterAdmin->getPropertyValues(new core_kernel_classes_Property(self::PROPERTY_ADMINISTRATOR_URI));
+        $proctorClass = $this->getRootClass();
+        $users = array();
+        foreach($testCenters as $testCenterUri){
+            $assignedProctors = $proctorClass->searchInstances(array(
+                self::PROPERTY_ASSIGNED_PROCTOR_URI => $testCenterUri
+            ), array(
+                'recursive' => true,
+                'like' => false
+            ));
+            $users = array_merge($users , $assignedProctors);
```

Comment:
as you can see here https://github.com/oat-sa/extension-tao-proctoring/pull/49/files#diff-d4d3195d505a239bb3cb856fd0a23e21R205 the array_merge should solve take care of it and return the proctor only once